File: c/ej3.c
#include <stdio.h>
#include <math.h>

// What is the largest prime factor of the number 600851475143?

int is_prime(int n){
    int a = floor(sqrt(n));
    for(int i = 2; i < a; i++){
        if(n % i == 0)
            return 0;
    }
    return 1;
}

int main(){
    unsigned long long b = 13195;
    unsigned long long largest_prime = 0;
    
    for(int i = 2; i < b; i++){
        if(b % i == 0){
            if(is_prime(i)){
                printf("%lu\n", i);
                largest_prime = i;
            }            
        }
    }
    printf("the largest prime factor is: %d\n", largest_prime);
    return 0;
}
    

File: python/ej3.py
import math

# What is the largest prime factor of the number 600851475143?

# Chequearemos el mayor factor primo solo hasta raiz de n, no hasta n en si

def is_prime(n):
    a = math.floor(math.sqrt(n))
    for i in range(2, a):
        if(n % i == 0):
            return 0
    return 1

def main():
    b = math.floor(600851475143 / 2) + 1
    largest_prime = 0
    for i in range(2, b):
        if(b % i == 0 and is_prime(i)):
            largest_prime = i
    print("The larges prime factor is: " + str(largest_prime))
    return

main()